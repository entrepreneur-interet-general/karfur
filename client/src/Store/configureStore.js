// Store/configureStore.js

import { createStore, applyMiddleware, combineReducers, compose } from 'redux';
import thunk from 'redux-thunk';

import langueReducer from './Reducers/langueReducer';
import dispositifReducer from './Reducers/dispositifReducer';
import structureReducer from './Reducers/structureReducer';
import userReducer from './Reducers/userReducer';
import ttsReducer from './Reducers/ttsReducer';

const rootReducer = combineReducers({langue: langueReducer, dispositif: dispositifReducer, user: userReducer, tts: ttsReducer, structure: structureReducer})

export default createStore(
  rootReducer, 
  compose(
    applyMiddleware(thunk),
    (window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()) || compose
  ))