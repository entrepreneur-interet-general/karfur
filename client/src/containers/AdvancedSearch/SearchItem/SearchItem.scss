@import 'src/scss/colors.scss';

$placeholderColor: $blanc;

.search-col{
  padding: 10px;
  display: flex;
  justify-content: center;
  align-items: center;

  .dropdown-menu.show{
    display: flex;
    box-shadow: 0px 4px 7px rgba(0, 0, 0, 0.2);
    background: $noir;
    border-radius: 12px;
    padding: 8px;
    max-width: 770px;
    z-index: 3;
    // flex-wrap: wrap;
  }

  .options-wrapper{
    margin-top:-10px;
    margin-bottom: -15px;
    max-height: 50vh;
    overflow: scroll;
    -ms-overflow-style: none;
    overflow: -moz-scrollbars-none;
    &::-webkit-scrollbar { width: 0 !important }
  }
  .search-options{
    white-space: nowrap;
    overflow: visible;
    margin-top:10px;

    &:not(:last-child){
      margin-right: 8px;
    }
  }
  .display-inline-block{
    display:inline-block;
  }
  .texte-blanc{
    color: #FFFFFF;
  }

  .search-dropdown{
    background: $gris;
    border-radius: 6px;
    padding: 20px;
  }

  .search-autocomplete{
    width: 150px;
    text-align: center;
    &::-webkit-input-placeholder {
      color: $placeholderColor;
      text-align: center;
    }
    &:-moz-placeholder {
      color: $placeholderColor;
      opacity: 1;
      text-align: center;
    }
    &::-moz-placeholder {
      color: $placeholderColor;
      opacity: 1;
      text-align: center;
    }
    &:-ms-input-placeholder {
      color: $placeholderColor;
      text-align: center;
    }
    &::-ms-input-placeholder {
      color: $placeholderColor;
      text-align: center;
    }
    &::placeholder {
      color: $placeholderColor;
      text-align: center;
    }
  }
}