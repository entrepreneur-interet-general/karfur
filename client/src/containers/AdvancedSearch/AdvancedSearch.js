import React, { Component } from 'react';
import { withTranslation } from 'react-i18next';
import track from 'react-tracking';
import { Col, Row, CardBody, CardFooter, Spinner, ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
import Swal from 'sweetalert2';
import querySearch from "stringquery";
import _ from "lodash";
import { NavHashLink } from 'react-router-hash-link';
import windowSize from 'react-window-size';
import { connect } from 'react-redux';
// import Cookies from 'js-cookie';

import SearchItem from './SearchItem/SearchItem';
import API from '../../utils/API';
import {initial_data} from "./data"
import CustomCard from '../../components/UI/CustomCard/CustomCard';
import EVAIcon from '../../components/UI/EVAIcon/EVAIcon';
import {filtres} from "../Dispositif/data";
import {filtres_contenu, tris} from "./data";
import {breakpoints} from 'utils/breakpoints.js';

import './AdvancedSearch.scss';
import variables from 'scss/colors.scss';

let user={_id:null, cookies:{}};
class AdvancedSearch extends Component {
  state = {
    showSpinner: false,
    recherche: initial_data.map( x => ({...x, active: false})),
    dispositifs: [],
    nbVues: [],
    pinned: [],
    activeFiltre: "",
    activeTri: "",
    data: [], //inutilisé, à remplacer par recherche quand les cookies sont stabilisés
    order: "created_at",
    croissant: true,
    filter:{},
    displayAll: true,
    dropdownOpenTri: false,
    dropdownOpenFiltre: false,
  }

  componentDidMount (){
    this.retrieveCookies();
    let tag=querySearch(this.props.location.search).tag;
    if(tag) { this.selectTag(decodeURIComponent(tag)) } else { this.queryDispositifs() }
    this._initializeEvents();
    window.scrollTo(0, 0);
  }

  queryDispositifs = (query=null) => {
    this.setState({ showSpinner: true })
    query = query || this.state.recherche.filter(x => x.active && x.queryName!=='localisation').map(x => (
      x.queryName === "audienceAge" ? 
      { "audienceAge.bottomValue": { $lt: x.topValue}, "audienceAge.topValue": { $gt: x.bottomValue} } :
      {[x.queryName]: x.query}
    )).reduce((acc, curr) => ({...acc, ...curr}),{});
    const localisationSearch = this.state.recherche.find(x => x.queryName === 'localisation' && x.value);
    API.get_dispositif({query: {...query, ...this.state.filter, status:'Actif', ...(!localisationSearch && {demarcheId: { $exists: false }}) }}).then(data_res => {
      let dispositifs=data_res.data.data;
      if(query["tags.name"]){       //On réarrange les résultats pour avoir les dispositifs dont le tag est le principal en premier
        dispositifs = dispositifs.sort((a,b)=> (a.tags.findIndex(x => x ? x.short === query["tags.name"] : 99) - b.tags.findIndex(x => x ? x.short === query["tags.name"] : 99)))
      }
      if(localisationSearch){       //On applique le filtre géographique maintenant
        dispositifs = dispositifs.filter(x => (
          x.typeContenu !== "demarche" || 
          !(x.variantes || []).some(y => y.villes) || 
          x.variantes.some(y => y.villes.some(z => (
            !z.address_components.some(ad => (
              !localisationSearch.query.some(lq => lq.long_name === ad.long_name) //On compare seulement les noms, il faudrait idéalement rajouter le type aussi mais la comparaison des Arrays me paraît lourde
            ))
          )))
        ));
        const filterDoubles = [...new Set(dispositifs.map(x => x.demarcheId || x._id))]; //Je vire les doublons créés par les variantes
        dispositifs = filterDoubles.map(x => dispositifs.find(y => y.demarcheId === x || y._id === x));
      }
      dispositifs = dispositifs.map(x => ({...x, nbVues: (this.state.nbVues.find(y=>y._id === x._id) || {}).count }))     //Je rajoute la donnée sur le nombre de vues par dispositif
        .filter(x => !this.state.pinned.some(y=>y._id===x._id || y===x._id))
      this.setState({ dispositifs:dispositifs, showSpinner: false })
    }).catch(()=>this.setState({ showSpinner: false }))
  }
  
  selectTag = (tag = {}) => {
    const tagValue = (filtres.tags.find(x => x.short === tag) || {});
    this.setState(pS => ({recherche: pS.recherche.map((x,i)=> i === 0 ? {...x, value: tagValue.name, short: tagValue.short, active: true} : x)}))
    this.queryDispositifs({"tags.short": tag})
    // this.props.history.replace("/advanced-search?tag="+tag)
  }
  
  _initializeEvents = () => {
    API.aggregate_events([
      {$match:
        {'action': 'readDispositif',
         'label': 'dispositifId',
         'value': { $ne: null } } },
      {$group:
         { _id : "$value",
          count:{ $sum: 1}}}
    ]).then(data_res => { 
      const countEvents = data_res.data.data;
      this.setState(pS => ({nbVues: countEvents, dispositifs: pS.dispositifs.map(x => ({...x, nbVues: (countEvents.find(y=>y._id === x._id) || {}).count })) }));
    })
  }

  retrieveCookies = () => {
    // Cookies.set('data', 'ici un test');
    // let dataC=Cookies.getJSON('data');
    // if(dataC){ this.setState({data:data.map((x,key)=> {return {...x, value:dataC[key] || x.value}})})}
    // let pinnedC=Cookies.getJSON('pinnedC');
    // if(pinnedC){ this.setState({pinned:pinnedC})}
    //data à changer en recherche après
    if(API.isAuth()){
      API.get_user_info().then(data_res => {
        let u=data_res.data.data;
        user={_id:u._id, cookies:u.cookies || {}}
        this.setState({
          pinned:user.cookies.parkourPinned || [],
          dispositifs:[...this.state.dispositifs].filter(x => !((user.cookies.parkourPinned || []).find( y=> y._id === x._id || y === x._id))),
          ...(user.cookies.parkourData && user.cookies.parkourData.length>0 && 
            {data:this.state.data.map((x,key)=> {return {...x, value: (user.cookies.parkourData[key] || x.value), query: (x.children.find(y=> y.name === (user.cookies.parkourData[key] || x.value)) || {}).query }})})
        })
      })
    }
  }

  pin = (e,dispositif) => {
    e.preventDefault();
    e.stopPropagation();
    dispositif.pinned=!dispositif.pinned;
    let prevState=[...this.state.dispositifs];
    this.setState({
      dispositifs: dispositif.pinned ? prevState.filter(x => x._id !== dispositif._id) : [...prevState,dispositif],
      pinned: dispositif.pinned ? 
        [...this.state.pinned, dispositif] :
        this.state.pinned.filter(x=> x._id !== dispositif._id)
    },()=>{
      user.cookies.parkourPinned=[...new Set(this.state.pinned.map(x => x._id))];
      API.set_user_info(user);
    })
  }

  reorder = tri => {
    console.log(tri, this.state.dispositifs);
    const order = tri.value,  croissant = order === this.state.order ? !this.state.croissant : true;
    this.setState(pS => ({dispositifs: pS.dispositifs.sort((a,b)=> {
      const aValue = _.get(a, order), bValue = _.get(b, order);
      return aValue > bValue ? (croissant ? 1 : -1) : aValue < bValue ? (croissant ? -1 : 1) : 0;
    }), order: tri.value, activeTri: tri.name, croissant: croissant}))
  }

  filter_content = filtre => {
    const filter = this.state.activeFiltre === filtre.name ? {} : filtre.query;
    const activeFiltre = this.state.activeFiltre === filtre.name ? "" : filtre.name;
    this.setState({filter, activeFiltre }, () => this.queryDispositifs());
  }

  goToDispositif = (dispositif={}, fromAutoSuggest=false) => {
    this.props.tracking.trackEvent({ action: 'click', label: 'goToDispositif' + (fromAutoSuggest ? ' - fromAutoSuggest' : ''), value : dispositif._id });
    this.props.history.push('/' + (dispositif.typeContenu || "dispositif") + (dispositif._id ? ('/' + dispositif._id) : ''))
  }

  upcoming = () => Swal.fire( {title:'Oh non!', text:'Cette fonctionnalité n\'est pas encore activée', type:'error', timer: 1500})

  selectParam = (key, subitem) => {
    let recherche = [...this.state.recherche];
    recherche[key]={
      ...recherche[key],
      value: subitem.name || subitem.formatted_address,
      query: subitem.query || subitem.address_components || subitem.name,
      active: true,
      ...(subitem.short && {short: subitem.short}),
      ...(subitem.bottomValue && {bottomValue: subitem.bottomValue}),
      ...(subitem.topValue && {topValue: subitem.topValue}),
    }
    this.setState({recherche: recherche}, ()=> this.queryDispositifs());
  }

  desactiver = key => this.setState({recherche: this.state.recherche.map((x, i) => i===key ? initial_data[i] : x)}, ()=> this.queryDispositifs());
  toggleDisplayAll = () => this.setState(pS => ({displayAll: !pS.displayAll}));
  toggleDropdownTri = () => this.setState(pS => ({ dropdownOpenTri: !pS.dropdownOpenTri }));
  toggleDropdownFiltre = () => this.setState(pS => ({ dropdownOpenFiltre: !pS.dropdownOpenFiltre }));

  render() {
    let {recherche, dispositifs, pinned, showSpinner, activeFiltre, activeTri, displayAll} = this.state;
    const {t, windowWidth, dispositifs: storeDispo} = this.props;
    const populatedPinned = storeDispo && storeDispo.length > 0 ? pinned.map(x => x._id ? x : (storeDispo.find(y => y._id === x) || {})) : [];
    const filteredPinned = activeFiltre ? populatedPinned.filter(x => activeFiltre === "Dispositifs" ? x.typeContenu !== "demarche" : x.typeContenu === "demarche") : populatedPinned;
    
    if(recherche[0].active){
      dispositifs = dispositifs.sort((a,b) => _.get(a,"tags.0.name", {}) === recherche[0].query ? -1 : _.get(b,"tags.0.name", {}) === recherche[0].query ? 1 : 0)
    }
    return (
      <div className="animated fadeIn advanced-search">
        <div className="search-bar">
          {recherche.filter((_,i) => displayAll || i===0 ).map((d,i) => (
            <SearchItem  
              key={i}
              item={d}
              keyValue={i}
              selectParam = {this.selectParam}
              desactiver={this.desactiver}
            />
          ))}
          <ResponsiveFooter
            {...this.state}
            show = {windowWidth < breakpoints.smLimit}
            toggleDropdownTri={this.toggleDropdownTri}
            toggleDropdownFiltre={this.toggleDropdownFiltre}
            reorder={this.reorder}
            filter_content={this.filter_content}
            toggleDisplayAll={this.toggleDisplayAll}
            t={t}
          />
        </div>
        <Row className="search-wrapper">
          {windowWidth >= breakpoints.smLimit && 
            <Col xl="2" lg="2" md="2" sm="2" xs="2" className="mt-250 side-col">
              {windowWidth >= breakpoints.desktopUp && 
                <EVAIcon name="options-2-outline" fill={variables.noir} className="mr-12" />}
              <div className="right-side">
                {windowWidth >= breakpoints.desktopUp ?
                  <b>{t("AdvancedSearch.Trier par", "Trier par :")}</b> :
                  <EVAIcon name="options-2-outline" fill={variables.noir} />}
                <div className="mt-10 side-options">
                  {tris.map((tri, idx) => (
                    <div 
                      key={idx} 
                      className={"side-option" + (tri.name === activeTri ? " active" : "")}
                      onClick={()=>this.reorder(tri)}
                    >
                      {t("AdvancedSearch." + tri.name, tri.name)}
                    </div>
                  ))}
                </div>
              </div>
            </Col>}
          <Col xl="8" lg="8" md="8" sm="8" xs="12" className="mt-250 central-col">
            <div className="results-wrapper">
              <Row>
                {[...filteredPinned,...dispositifs].map((dispositif) => {
                  if(!dispositif.hidden){
                    let shortTag = null;
                    if(dispositif.tags && dispositif.tags.length > 0 && dispositif.tags[0] && dispositif.tags[0].short){ shortTag = (dispositif.tags[0].short || {}).replace(/ /g, "-") }
                    return (
                      <Col xl="3" lg="3" md="4" sm="6" xs="12" className={"card-col puff-in-center " + (dispositif.typeContenu || "dispositif")} key={dispositif._id}>
                        <CustomCard onClick={() => this.goToDispositif(dispositif)} className={(dispositif.typeContenu === "demarche" ? ("texte-" + shortTag + " bg-light-" + shortTag) : "")}>
                          <CardBody>
                            <EVAIcon 
                              name="bookmark" 
                              size="xlarge" 
                              onClick={(e)=>this.pin(e,dispositif)} 
                              fill={(dispositif.pinned ? variables.noir : variables.noirCD)} 
                              className={"bookmark-icon" + (dispositif.pinned ? " pinned":"")} 
                            />
                            <h5>{dispositif.titreInformatif}</h5>
                            <p>{dispositif.abstract}</p>
                          </CardBody>
                          {dispositif.typeContenu !== "demarche" &&
                            <CardFooter className={"correct-radius align-right bg-" + shortTag}>{dispositif.titreMarque}</CardFooter>}
                        </CustomCard>
                      </Col>
                    )
                  }else{
                    return false
                  }}
                )}
                {!showSpinner && [...pinned,...dispositifs].length === 0 &&
                  <Col xs="12" sm="6" md="3" className="no-result" onClick={()=>this.selectTag()}>
                    <CustomCard className="no-result-card">
                      <CardBody>
                        <h5>{t("Aucun résultat", "Aucun résultat")}</h5>
                        <p>{t("AdvancedSearch.Elargir recherche", "Essayez d’élargir votre recherche en désactivant certains filtres")} </p>
                      </CardBody>
                      <CardFooter className="align-right">
                        {t("AdvancedSearch.Oups", "Oups !")}...
                      </CardFooter>
                    </CustomCard>
                  </Col>}
                <Col xs="12" sm="6" md="3">
                  <NavHashLink to="/comment-contribuer#ecrire">
                    <CustomCard addcard="true" className="plus-card">
                      <CardBody>
                        {showSpinner ?
                          <Spinner color="success" /> : 
                          <span className="add-sign">+</span> }
                      </CardBody>
                      <CardFooter className="align-right">
                        {showSpinner ? t("Chargement", "Chargement") + "..." : t("AdvancedSearch.Créer une fiche", "Créer une fiche")}
                      </CardFooter>
                    </CustomCard>
                  </NavHashLink>
                </Col>
              </Row>
            </div>
          </Col>
          {windowWidth >= breakpoints.smLimit && 
            <Col xl="2" lg="2" md="2" sm="2" xs="2" className="mt-250 side-col">
              {windowWidth >= breakpoints.desktopUp && 
                <EVAIcon name="funnel-outline" fill={variables.noir} className="mr-12" />}
              <div className="right-side">
                {windowWidth >= breakpoints.desktopUp ?
                  <b>{t("AdvancedSearch.Filtrer par", "Filtrer par :")}</b> :
                  <EVAIcon name="funnel-outline" fill={variables.noir} />}
                <div className="mt-10 side-options">
                  {filtres_contenu.map((filtre, idx) => (
                    <div 
                      key={idx} 
                      className={"side-option right" + (filtre.name === activeFiltre ? " active" : "")}
                      onClick={()=>this.filter_content(filtre)}
                    >
                      {filtre.name && t("AdvancedSearch." + filtre.name, filtre.name)}
                    </div>
                  ))}
                </div>
              </div>
            </Col>}
        </Row>
      </div>
    )
  }
}

const ResponsiveFooter = props => {
  const {activeFiltre, activeTri, displayAll, t, show} = props;
  return show && (
    <div className="responsive-footer">
      <ButtonDropdown className={"options-dropdown" + (activeTri ? " active" : "")} isOpen={props.dropdownOpenTri} toggle={props.toggleDropdownTri}>
        <DropdownToggle color="transparent">
          <EVAIcon name="options-2-outline" />
        </DropdownToggle>
        <DropdownMenu>
          {tris.map((tri, idx) => (
            <DropdownItem 
              key={idx} 
              onClick={()=>props.reorder(tri)} 
              className={"side-option" + (tri.name === activeTri ? " active" : "")}
            >
              {t("AdvancedSearch." + tri.name, tri.name)}
            </DropdownItem>
          ))}
        </DropdownMenu>
      </ButtonDropdown>
      <ButtonDropdown className={"options-dropdown" + (activeFiltre ? " active" : "")} isOpen={props.dropdownOpenFiltre} toggle={props.toggleDropdownFiltre}>
        <DropdownToggle color="transparent" className={activeFiltre ? "active" : ""}>
          <EVAIcon name="funnel-outline" />
        </DropdownToggle>
        <DropdownMenu>
          {filtres_contenu.map((filtre, idx) => (
            <DropdownItem 
              key={idx} 
              onClick={()=>props.filter_content(filtre)} 
              className={"side-option" + (filtre.name === activeFiltre ? " active" : "")}
            >
              {filtre.name && t("AdvancedSearch." + filtre.name, filtre.name)}
            </DropdownItem>
          ))}
        </DropdownMenu>
      </ButtonDropdown>
      <EVAIcon 
        name={"arrow-circle-" + (displayAll ? "up" : "down")} 
        size="xlarge" 
        onClick={props.toggleDisplayAll} 
        className="close-arrow" 
        fill={variables.grisFonce} 
      />
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    dispositifs: state.dispositif.dispositifs,
  }
}

export default track({
    page: 'AdvancedSearch',
  })(
    connect(mapStateToProps)(
      withTranslation()(
        windowSize(AdvancedSearch)
      )
    )
  );

