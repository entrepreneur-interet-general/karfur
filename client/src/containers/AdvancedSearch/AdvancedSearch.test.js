import React from 'react';
import {AdvancedSearch, ResponsiveFooter} from './AdvancedSearch';
import { expect } from 'chai';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import sinon from 'sinon';
import { CardFooter} from 'reactstrap';

import SearchItem from '../AdvancedSearch/SearchItem/SearchItem';
import {breakpoints} from 'utils/breakpoints.js';

const mockStore = configureStore();
const store = mockStore({});
const location = {search:""};

describe('AdvancedSearch', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Provider store={store}><AdvancedSearch t={k=>k} location={location} /></Provider> ).dive().dive();
  })
  
  it('renders without crashing', () => {
    expect(wrapper.is('div')).to.equal(true);
    expect(wrapper.find(CardFooter).childAt(0).text()).to.equal("Chargement...");
  });

  it('calls componentDidMount', () => {
    sinon.spy(AdvancedSearch.prototype, 'componentDidMount');
    shallow(<Provider store={store}><AdvancedSearch t={k=>k} location={location} /></Provider> ).dive().dive();
    expect(AdvancedSearch.prototype.componentDidMount).to.have.property('callCount', 1);
    AdvancedSearch.prototype.componentDidMount.restore();
  });

  it('should match its reference snapshot', () => {
    expect(wrapper).to.matchSnapshot()
  });

  it('contains AdvancedSearch in className', () => {
    expect(wrapper.props().className).to.contain("advanced-search");
  });

  it('should have an empty dispositifs list in state', () => {
    expect(wrapper.state().dispositifs).to.be.an('array').that.is.empty;
  });

  it('renders children when passed in', () => {
    const wrapper = shallow((
      <Provider store={store}>
        <AdvancedSearch t={k=>k} location={location} >
          <div className="unique" />
        </AdvancedSearch>
      </Provider>
    ));
    expect(wrapper.contains(<div className="unique" />)).to.equal(true);
  });

  it('contains SearchItems', () => {
    expect(wrapper.containsMatchingElement(SearchItem)).to.equal(true);
  });

  it('should have 4 SearchItems', () => {
    expect(wrapper.find(SearchItem)).to.have.lengthOf(4);
  });

  it('should not show ResponsiveFooter by default', () => {
    expect(wrapper.find(ResponsiveFooter)).to.have.lengthOf(1);
    expect(wrapper.find(ResponsiveFooter).props("show")).to.be.false;
  });

  it('should show ResponsiveFooter on small screens', () => {
    wrapper.setProps({windowWidth: breakpoints.smLimit - 1})
    expect(wrapper.find(ResponsiveFooter).props("show")).to.be.true;
  });
})

