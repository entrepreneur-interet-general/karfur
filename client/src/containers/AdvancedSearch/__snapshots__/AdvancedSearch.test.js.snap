// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvancedSearch should match its reference snapshot 1`] = `
<div
  className="animated fadeIn advanced-search"
>
  <div
    className="search-bar"
  >
    <withI18nextTranslation(SearchItem)
      desactiver={[Function]}
      item={
        Object {
          "active": false,
          "children": Array [
            Object {
              "darkColor": "#3E2884",
              "hoverColor": "#3E2884",
              "illustrationColor": "#F9AA38",
              "lightColor": "#F0E8F5",
              "name": "apprendre le français",
              "short": "Français",
            },
            Object {
              "darkColor": "#188CC6",
              "hoverColor": "#188CC6",
              "illustrationColor": "#F77B0B",
              "lightColor": "#DDF3FA",
              "name": "me loger",
              "short": "Logement",
            },
            Object {
              "darkColor": "#149295",
              "hoverColor": "#4FD3BD",
              "illustrationColor": "#DD0539",
              "lightColor": "#D6EFF4",
              "name": "trouver un travail",
              "short": "Insertion pro",
            },
            Object {
              "darkColor": "#137F3A",
              "hoverColor": "#7DC690",
              "illustrationColor": "#FC1E73",
              "lightColor": "#E8F4E9",
              "name": "apprendre un métier",
              "short": "Formation pro",
            },
            Object {
              "darkColor": "#307205",
              "hoverColor": "#ADEA5C",
              "illustrationColor": "#E561D8",
              "lightColor": "#E8F7CD",
              "name": "faire des études",
              "short": "Études",
            },
            Object {
              "darkColor": "#F97821",
              "hoverColor": "#F9D1B7",
              "illustrationColor": "#5435D6",
              "lightColor": "#FFF0E7",
              "name": "me déplacer",
              "short": "Mobilité",
            },
            Object {
              "darkColor": "#0A54BF",
              "hoverColor": "#FDA8AD",
              "illustrationColor": "#1FC2C1",
              "lightColor": "#FFEEEE",
              "name": "aider une association",
              "short": "Bénévolat",
            },
            Object {
              "darkColor": "#8E1B5A",
              "hoverColor": "#FDA5C4",
              "illustrationColor": "#76D327",
              "lightColor": "#FFEBF5",
              "name": "rencontrer des gens",
              "short": "Rencontre",
            },
            Object {
              "darkColor": "#861B8E",
              "hoverColor": "#E595F9",
              "illustrationColor": "#FCBF35",
              "lightColor": "#F6E6FC",
              "name": "occuper mon temps libre",
              "short": "Loisir",
            },
            Object {
              "darkColor": "#ED4C26",
              "hoverColor": "#FEC396",
              "illustrationColor": "#1898FC",
              "lightColor": "#FFF0E7",
              "name": "découvrir la culture",
              "short": "Culture",
            },
            Object {
              "darkColor": "#C6093F",
              "hoverColor": "#F7698E",
              "illustrationColor": "#1FC2C1",
              "lightColor": "#FFEEEE",
              "name": "me soigner",
              "short": "Santé",
            },
          ],
          "placeholder": "choisir un thème",
          "query": "Apprendre le français",
          "queryName": "tags.name",
          "title": "Je cherche à",
          "value": null,
        }
      }
      key="0"
      keyValue={0}
      selectParam={[Function]}
    />
    <withI18nextTranslation(SearchItem)
      desactiver={[Function]}
      item={
        Object {
          "active": false,
          "placeholder": "ma ville",
          "queryName": "localisation",
          "title": "J'habite à",
          "value": null,
        }
      }
      key="1"
      keyValue={1}
      selectParam={[Function]}
    />
    <withI18nextTranslation(SearchItem)
      desactiver={[Function]}
      item={
        Object {
          "active": false,
          "children": Array [
            Object {
              "bottomValue": 0,
              "name": "moins de 18 ans",
              "topValue": 18,
            },
            Object {
              "bottomValue": 18,
              "name": "entre 18 et 25 ans",
              "topValue": 25,
            },
            Object {
              "bottomValue": 25,
              "name": "entre 25 et 56 ans",
              "topValue": 56,
            },
            Object {
              "bottomValue": 56,
              "name": "56 ans et plus",
              "topValue": 120,
            },
          ],
          "placeholder": "âge",
          "queryName": "audienceAge",
          "title": "J'ai",
          "value": null,
        }
      }
      key="2"
      keyValue={2}
      selectParam={[Function]}
    />
    <withI18nextTranslation(SearchItem)
      desactiver={[Function]}
      item={
        Object {
          "active": false,
          "append": "Quel est mon niveau ?",
          "children": Array [
            Object {
              "name": "pas du tout",
              "query": "Débutant",
            },
            Object {
              "name": "un peu",
              "query": "Débutant",
            },
            Object {
              "name": "moyennement",
              "query": "Intermédiaire",
            },
            Object {
              "name": "bien",
              "query": "Intermédiaire",
            },
            Object {
              "name": "très bien",
              "query": "Avancé",
            },
          ],
          "placeholder": "mon niveau",
          "queryName": "niveauFrancais",
          "title": "Je parle",
          "title2": "français",
          "value": null,
        }
      }
      key="3"
      keyValue={3}
      selectParam={[Function]}
    />
    <ResponsiveFooter
      activeFiltre=""
      activeTri=""
      croissant={true}
      data={Array []}
      displayAll={true}
      dispositifs={Array []}
      dropdownOpenFiltre={false}
      dropdownOpenTri={false}
      filter={Object {}}
      filter_content={[Function]}
      nbVues={Array []}
      order="created_at"
      pinned={Array []}
      recherche={
        Array [
          Object {
            "active": false,
            "children": Array [
              Object {
                "darkColor": "#3E2884",
                "hoverColor": "#3E2884",
                "illustrationColor": "#F9AA38",
                "lightColor": "#F0E8F5",
                "name": "apprendre le français",
                "short": "Français",
              },
              Object {
                "darkColor": "#188CC6",
                "hoverColor": "#188CC6",
                "illustrationColor": "#F77B0B",
                "lightColor": "#DDF3FA",
                "name": "me loger",
                "short": "Logement",
              },
              Object {
                "darkColor": "#149295",
                "hoverColor": "#4FD3BD",
                "illustrationColor": "#DD0539",
                "lightColor": "#D6EFF4",
                "name": "trouver un travail",
                "short": "Insertion pro",
              },
              Object {
                "darkColor": "#137F3A",
                "hoverColor": "#7DC690",
                "illustrationColor": "#FC1E73",
                "lightColor": "#E8F4E9",
                "name": "apprendre un métier",
                "short": "Formation pro",
              },
              Object {
                "darkColor": "#307205",
                "hoverColor": "#ADEA5C",
                "illustrationColor": "#E561D8",
                "lightColor": "#E8F7CD",
                "name": "faire des études",
                "short": "Études",
              },
              Object {
                "darkColor": "#F97821",
                "hoverColor": "#F9D1B7",
                "illustrationColor": "#5435D6",
                "lightColor": "#FFF0E7",
                "name": "me déplacer",
                "short": "Mobilité",
              },
              Object {
                "darkColor": "#0A54BF",
                "hoverColor": "#FDA8AD",
                "illustrationColor": "#1FC2C1",
                "lightColor": "#FFEEEE",
                "name": "aider une association",
                "short": "Bénévolat",
              },
              Object {
                "darkColor": "#8E1B5A",
                "hoverColor": "#FDA5C4",
                "illustrationColor": "#76D327",
                "lightColor": "#FFEBF5",
                "name": "rencontrer des gens",
                "short": "Rencontre",
              },
              Object {
                "darkColor": "#861B8E",
                "hoverColor": "#E595F9",
                "illustrationColor": "#FCBF35",
                "lightColor": "#F6E6FC",
                "name": "occuper mon temps libre",
                "short": "Loisir",
              },
              Object {
                "darkColor": "#ED4C26",
                "hoverColor": "#FEC396",
                "illustrationColor": "#1898FC",
                "lightColor": "#FFF0E7",
                "name": "découvrir la culture",
                "short": "Culture",
              },
              Object {
                "darkColor": "#C6093F",
                "hoverColor": "#F7698E",
                "illustrationColor": "#1FC2C1",
                "lightColor": "#FFEEEE",
                "name": "me soigner",
                "short": "Santé",
              },
            ],
            "placeholder": "choisir un thème",
            "query": "Apprendre le français",
            "queryName": "tags.name",
            "title": "Je cherche à",
            "value": null,
          },
          Object {
            "active": false,
            "placeholder": "ma ville",
            "queryName": "localisation",
            "title": "J'habite à",
            "value": null,
          },
          Object {
            "active": false,
            "children": Array [
              Object {
                "bottomValue": 0,
                "name": "moins de 18 ans",
                "topValue": 18,
              },
              Object {
                "bottomValue": 18,
                "name": "entre 18 et 25 ans",
                "topValue": 25,
              },
              Object {
                "bottomValue": 25,
                "name": "entre 25 et 56 ans",
                "topValue": 56,
              },
              Object {
                "bottomValue": 56,
                "name": "56 ans et plus",
                "topValue": 120,
              },
            ],
            "placeholder": "âge",
            "queryName": "audienceAge",
            "title": "J'ai",
            "value": null,
          },
          Object {
            "active": false,
            "append": "Quel est mon niveau ?",
            "children": Array [
              Object {
                "name": "pas du tout",
                "query": "Débutant",
              },
              Object {
                "name": "un peu",
                "query": "Débutant",
              },
              Object {
                "name": "moyennement",
                "query": "Intermédiaire",
              },
              Object {
                "name": "bien",
                "query": "Intermédiaire",
              },
              Object {
                "name": "très bien",
                "query": "Avancé",
              },
            ],
            "placeholder": "mon niveau",
            "queryName": "niveauFrancais",
            "title": "Je parle",
            "title2": "français",
            "value": null,
          },
        ]
      }
      reorder={[Function]}
      show={false}
      showSpinner={true}
      t={[Function]}
      toggleDisplayAll={[Function]}
      toggleDropdownFiltre={[Function]}
      toggleDropdownTri={[Function]}
    />
  </div>
  <Row
    className="search-wrapper"
    tag="div"
  >
    <Col
      className="mt-250 central-col"
      lg="8"
      md="8"
      sm="8"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xl="8"
      xs="12"
    >
      <div
        className="results-wrapper"
      >
        <Row
          tag="div"
        >
          <Col
            md="3"
            sm="6"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="12"
          >
            <NavHashLink
              to="/comment-contribuer#ecrire"
            >
              <customCard
                addcard="true"
                className="create-card"
              >
                <CardBody
                  tag="div"
                >
                  <Spinner
                    color="success"
                    tag="div"
                    type="border"
                  >
                    Loading...
                  </Spinner>
                </CardBody>
                <CardFooter
                  className="align-right"
                  tag="div"
                >
                  Chargement...
                </CardFooter>
              </customCard>
            </NavHashLink>
          </Col>
        </Row>
      </div>
    </Col>
  </Row>
</div>
`;
